src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1086,6 +1086,46 @@ public void TestConnectionStringEnvVarHasHighestPrecedence()
             }
         }
 
+        /// <summary>
+        /// Test to validate that when an entity which will return a paginated response is queried and a custom rest base route is configured in the runtime configuration,
+        /// then the generated nextLink in the response would contain the rest base-route just before the rest path. For the subsequent query, the rest base-route will be trimmed
+        /// by the upstream before the request lands at DAB.
+        /// </summary>
+        [TestMethod]
+        public async Task TestRestBaseRouteInNextLinkForPaginatedRestResponse()
+        {
+            const string CUSTOM_CONFIG = "custom-config.json";
+            string restBaseRoute = "/rest-api";
+            TestHelper.ConstructNewConfigWithCustomSettings(CUSTOM_CONFIG, TestCategory.MSSQL, restBaseRoute: restBaseRoute);
+            string[] args = new[]
+            {
+                    $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                string requestPath = "/api/MappedBookmarks";
+                HttpMethod httpMethod = SqlTestHelper.ConvertRestMethodToHttpMethod(RestMethod.Get);
+                HttpRequestMessage request = new(httpMethod, requestPath);
+
+                HttpResponseMessage response = await client.SendAsync(request);
+                string responseBody = await response.Content.ReadAsStringAsync();
+                Assert.IsTrue(response.StatusCode is HttpStatusCode.OK);
+
+                JsonElement responseElement = JsonSerializer.Deserialize<JsonElement>(responseBody);
+                JsonElement responseValue = responseElement.GetProperty(SqlTestHelper.jsonResultTopLevelKey);
+                string nextLink = responseElement.GetProperty("nextLink").ToString();
+
+                // Assert that we got an array response with length equal to the maximum allowed records in a paginated response.
+                Assert.IsTrue(responseValue.ValueKind is JsonValueKind.Array);
+                Assert.IsTrue(responseValue.GetArrayLength() == 100);
```

Comment:
```suggestion
                Assert.AreEqual(100, responseValue.GetArrayLength());
```

Let's use `AreEqual` rather than `IsTrue` with an assert so it's more obvious when it fails.